### ISSUE TITLE
Missing Weapon Bug on specific maps/modes

### ISSUE THREAD

**Issue by lywit** [Joe91/fun-bots#342]
In this thread I will attempt to track on which maps the USE_LOAD_BUNDLE_BUGFIX does not work to fix the missing weapons bug with the latest version of Funbots .

So far someone reported through Discord that Damavand Peak on the Rush gamemode still has the issue despite using the fix.

Maps not working at the moment:
- Damavand Peak

**Comment by Joe91** [comment]
Thanks a lot for starting this! 
Do we have an easy way to test for this behavior or is it still just playing with a huge amount of bots?

**Comment by lywit** [comment]
> Thanks a lot for starting this! 
> Do we have an easy way to test for this behavior or is it still just playing with a huge amount of bots?

Yes, right now the best method I've found is to run a server with as many bots as possible, but it can still take a while to occur.

Decreasing spawn timers can help :)

Always validate without the load bundle fix first, so that the bug definitely occurs with your server config.

**Comment by Mei99-CN** [comment]
I get this bug too. 
The bug only occur in the maps with vehicles.
When the bug occur,  the VU console will print lots of logs as below:

 [2025-05-09 15:17:47+08:00] [error] [VeniceEXT] [fun-bots-2.9.3-release] Error: [string "Vehicles.lua"]:53: cannot cast an instance of type 'SoldierEntityData' to 'VehicleEntityData'
stack traceback:
	[C]: in global 'VehicleEntityData'
	[string "Vehicles.lua"]:53: in method 'GetVehicleByEntity'
	[string "GameDirector.lua"]:737: in method 'OnVehicleExit'
	[string "__init__.lua"]:521: in function <[string "__init__.lua"]:520>
	[C]: in method 'ExitVehicle'
	[string "Bot.lua"]:293: in method '_DoExitVehicle'
	[string "BotUpdate.lua"]:190: in method 'OnVehicleFastTimerUpdate'
	[string "BotUpdate.lua"]:116: in method 'FastTimerUpdate'
	[string "BotUpdate.lua"]:69: in method 'OnUpdatePassPostFrame'
	[string "BotManager.lua"]:78: in method 'OnUpdateManagerUpdate'
	[string "__init__.lua"]:259: in function <[string "__init__.lua"]:258>
[2025-05-09 15:20:41+08:00] [error] [VeniceEXT] [fun-bots-2.9.3-release] Error: [string "Vehicles.lua"]:53: cannot cast an instance of type 'SoldierEntityData' to 'VehicleEntityData'
stack traceback:
	[C]: in global 'VehicleEntityData'
	[string "Vehicles.lua"]:53: in method 'GetVehicleByEntity'
	[string "GameDirector.lua"]:737: in method 'OnVehicleExit'
	[string "__init__.lua"]:521: in function <[string "__init__.lua"]:520>
	[C]: in method 'ExitVehicle'
	[string "Bot.lua"]:293: in method '_DoExitVehicle'
	[string "BotUpdate.lua"]:190: in method 'OnVehicleFastTimerUpdate'
	[string "BotUpdate.lua"]:116: in method 'FastTimerUpdate'
	[string "BotUpdate.lua"]:69: in method 'OnUpdatePassPostFrame'
	[string "BotManager.lua"]:78: in method 'OnUpdateManagerUpdate'
	[string "__init__.lua"]:259: in function <[string "__init__.lua"]:258>
[2025-05-09 15:27:49+08:00] [error] [VeniceEXT] [fun-bots-2.9.3-release] Error: [string "Vehicles.lua"]:53: cannot cast an instance of type 'SoldierEntityData' to 'VehicleEntityData'
stack traceback:
	[C]: in global 'VehicleEntityData'
	[string "Vehicles.lua"]:53: in method 'GetVehicleByEntity'
	[string "GameDirector.lua"]:737: in method 'OnVehicleExit'
	[string "__init__.lua"]:521: in function <[string "__init__.lua"]:520>
	[C]: in method 'ExitVehicle'
	[string "Bot.lua"]:293: in method '_DoExitVehicle'
	[string "BotUpdate.lua"]:190: in method 'OnVehicleFastTimerUpdate'
	[string "BotUpdate.lua"]:116: in method 'FastTimerUpdate'
	[string "BotUpdate.lua"]:69: in method 'OnUpdatePassPostFrame'
	[string "BotManager.lua"]:78: in method 'OnUpdateManagerUpdate'
	[string "__init__.lua"]:259: in function <[string "__init__.lua"]:258>
[2025-05-09 15:35:21+08:00] [error] [VeniceEXT] [fun-bots-2.9.3-release] Error: [string "Bot.lua"]:519: attempt to index a nil value (field 'controlledControllable')
stack traceback:
	[string "Bot.lua"]:519: in method 'ShootAt'
	[string "BotManager.lua"]:278: in method 'OnShootAt'
	[string "BotManager.lua"]:238: in method 'OnSoldierDamage'
	[string "__init__.lua"]:535: in function <[string "__init__.lua"]:534>
[2025-05-09 15:35:43+08:00] [error] [VeniceEXT] [fun-bots-2.9.3-release] Error: [string "Bot.lua"]:519: attempt to index a nil value (field 'controlledControllable')
stack traceback:
	[string "Bot.lua"]:519: in method 'ShootAt'
	[string "BotManager.lua"]:278: in method 'OnShootAt'
	[string "BotManager.lua"]:238: in method 'OnSoldierDamage'
	[string "__init__.lua"]:535: in function <[string "__init__.lua"]:534>
[2025-05-09 15:38:24+08:00] [error] [VeniceEXT] [fun-bots-2.9.3-release] Error: [string "Vehicles.lua"]:53: cannot cast an instance of type 'SoldierEntityData' to 'VehicleEntityData'
stack traceback:
	[C]: in global 'VehicleEntityData'
	[string "Vehicles.lua"]:53: in method 'GetVehicleByEntity'
	[string "GameDirector.lua"]:737: in method 'OnVehicleExit'
	[string "__init__.lua"]:521: in function <[string "__init__.lua"]:520>
	[C]: in method 'ExitVehicle'
	[string "Bot.lua"]:293: in method '_DoExitVehicle'
	[string "BotUpdate.lua"]:190: in method 'OnVehicleFastTimerUpdate'
	[string "BotUpdate.lua"]:116: in method 'FastTimerUpdate'
	[string "BotUpdate.lua"]:69: in method 'OnUpdatePassPostFrame'
	[string "BotManager.lua"]:78: in method 'OnUpdateManagerUpdate'
	[string "__init__.lua"]:259: in function <[string "__init__.lua"]:258>

**Comment by Mei99-CN** [comment]
And the bug might be triggered more frequently after the respawn after death on vehicle, with the log below:
[2025-05-09 22:34:22+08:00] [info] [VeniceEXT] [fun-bots-2.9.3-release] !!! was nil !!!
[2025-05-09 22:34:24+08:00] [info] [VeniceEXT] [fun-bots-2.9.3-release] !!! was nil !!!
[2025-05-09 22:35:46+08:00] [info] [VeniceEXT] [fun-bots-2.9.3-release] !!! was nil !!!
[2025-05-09 22:36:48+08:00] [info] [VeniceEXT] [fun-bots-2.9.3-release] !!! was nil !!!
[2025-05-09 22:37:49+08:00] [info] [VeniceEXT] [fun-bots-2.9.3-release] !!! was nil !!!
[2025-05-09 22:39:32+08:00] [info] [VeniceEXT] [fun-bots-2.9.3-release] !!! was nil !!!
[2025-05-09 22:40:34+08:00] [info] [VeniceEXT] [fun-bots-2.9.3-release] !!! was nil !!!
[2025-05-09 22:41:37+08:00] [info] [VeniceEXT] [fun-bots-2.9.3-release] !!! was nil !!!
[2025-05-09 22:43:20+08:00] [info] [VeniceEXT] [fun-bots-2.9.3-release] !!! was nil !!!

**Comment by Joe91** [comment]
If you have the same issue, try to change this line to ` USE_LOAD_BUNDLE_BUGFIX = true,`

https://github.com/Joe91/fun-bots/blob/master/ext/Shared/Registry/Registry.lua#L27

(this is also related to this issue: related also to this issue:
https://github.com/Joe91/fun-bots/issues/286)
